submissions/factorio_draftsman/main.md

```diff
@@ -0,0 +1,200 @@
+* **Draftsman supports mods.** Draftsman emulates the [Factorio data lifecycle](https://lua-api.factorio.com/latest/Data-Lifecycle.html) directly, meaning that the same loading process that happens when you launch the game is mimicked with a single Draftsman function. In addition to ensuring absolute accuracy to Factorio, this also makes it compatible with custom mod prototypes that can be accessed in script as if they were any other vanilla entity.
+
+Draftsman also has support for custom `"EntityLike"` objects, most notably `Group` objects, that allow you to create custom constructs that can be inserted into blueprints for aid in clarity and compartmentalization. Again, *MAYBE* I got a little carried away.
+
+In an effort to keep this document brief, I'm not going to go into the nitty-gritty about the module or how exactly it works; I've spent a lot of time writing documentation for that purpose instead. Rather, I'm going to show off a couple of things that I've made with it so far, as well as potential things that could be made in the future, to try to illustrate exactly why I spent all this time in the first place. 
+
+## Automatic Item Stack Sizes
+
+Oftentimes for you want to figure out how much storage you need for an amount of an item. However, storage in Factorio is based on slots, not amounts, so the amount of storage is actually dependent on not only the amount you're trying to store but also the stack size of that item. [You can design a circuit contraption](https://factorioprints.com/view/-LwH68dlvOEIKWkU5JqR) to determine this amount of slots that you need, but you need a big cell of combinators documenting the item sizes. This is *boring* to make, and easily broken if a new item is added or the stack size changes. Clearly, for such a simple and repetitive task, a script is better suited:
+
+```python
+# Create a N x 5 grid of connected constant combinators, with every item and their stack size
+
+from draftsman.blueprintable import Blueprint
+from draftsman.constants import Direction
+from draftsman.data import items
+from draftsman.entity import ConstantCombinator
+
+
+def main():
+    blueprint = Blueprint()
+
+    count = 0  # Total number of signals
+    index = 0  # Signal index in the current combinator
+    i = 0  # How many combinators we've added
+    x = 0
+    y = 0
+    combinator = ConstantCombinator(direction=Direction.SOUTH)
+
+    # Iterate over every item in order:
+    for item in items.raw:
+        # Ignore hidden items/entities
+        if "flags" in items.raw[item]:
+            if "hidden" in items.raw[item]["flags"]:
+                continue
+        # Keep track of how many signals we've gone through
+        count += 1
+        # Write the stack size signal
+        stack_size = items.raw[item]["stack_size"]
+        combinator.set_signal(index, item, stack_size)
+        index += 1
+        # Once we exceed the current combinator, place it and reset
+        if index == 20:
```

Comment:
Can you add a constant for 20 in case there is some significance to the number? I presume it is how many signals can be defined in a constant combinator, but I haven't really used combinators much. `signals_per_combinator` or `signals_per_constant_combinator`?

Reply:
Changed to:
```python
if signal_index == combinator.item_slot_count:
    # ...
```
The 20 was leftover from when the item_slot_count of constant combinators wasn't extracted. Now it can be used and is dynamic to whichever combinator was specified in the beginning.